wagtail/blocks/field_block.py

```diff
@@ -160,7 +162,7 @@ def __init__(
         super().__init__(**kwargs)
 
     def get_searchable_content(self, value):
-        return [force_str(value)]
+        return [force_str(value)] if self.search_index is True else []
```

Comment:
It's best to avoid `is True` or `is False` in tests like this - Python has well-established rules about which values behave as true or false, such as `1` being true and `None` being false. Writing `if self.search_index is True` means that _only_ the exact value `True` will be recognised, and a definition like `CharBlock(search_index=1)` won't work as expected. To be fair, it's unlikely that a developer would pass anything other than `True` or `False` in this case, but it's best to get into the habit of writing boolean tests in a way that works for any true or false value:

```suggestion
        return [force_str(value)] if self.search_index else []
```